metaflow/metaflow_config_funcs.py

```diff
@@ -106,6 +106,28 @@ def from_conf(name, default=None, validate_fn=None):
     return value
 
 
+def from_conf_dict(prefix, validate_fn=None):
```

Comment:
@zendesk-klross Any specific reason for this change? Given that this PR focuses on just the encryption bits - I wonder if we need to support arbitrary headers?

Reply:
Happy to jump on a quick call if that would be useful to get to a quick conclusion.

Reply:
We added the encryption attribute to S3 objects mainly to test a use-case for the new custom headers (this way, we can make sure the object actually gets updated using these headers). But we also figured it's a good idea to let users pick their own headers for their specific needs, as mentioned [here](https://github.com/Netflix/metaflow/issues/1191).

Reply:
We need to ensure that we take care of all scenarios with arbitrary headers, including any considerations that would affect the data being read back. If you have a specific use case, we are happy to tackle that as part of this PR.

Reply:
We also thought that this parsing function might be useful for other configurable arguments that we don't want to explicitly define in Metaflow, but let us know if you can't think of any other applications for this.

Reply:
Right now the PR starts the groundwork to support all these options: `ALLOWED_UPLOAD_ARGS = ['ACL', 'CacheControl', 'ChecksumAlgorithm', 'ContentDisposition', 'ContentEncoding', 'ContentLanguage', 'ContentType', 'ExpectedBucketOwner', 'Expires', 'GrantFullControl', 'GrantRead', 'GrantReadACP', 'GrantWriteACP', 'Metadata', 'ObjectLockLegalHoldStatus', 'ObjectLockMode', 'ObjectLockRetainUntilDate', 'RequestPayer', 'ServerSideEncryption', 'StorageClass', 'SSECustomerAlgorithm', 'SSECustomerKey', 'SSECustomerKeyMD5', 'SSEKMSKeyId', 'SSEKMSEncryptionContext', 'Tagging', 'WebsiteRedirectLocation'].` The list is pretty long and testing for all combinations of headers might be complicated, so if you feel that an untested setup isn't a good idea we can definitely refocus the PR to support ServerSideEncryption only.

Reply:
We're also happy to expand this PR to include support for these `ALLOWED_DOWNLOAD_ARGS = ['ChecksumMode', 'VersionId', 'SSECustomerAlgorithm', 'SSECustomerKey', 'SSECustomerKeyMD5', 'RequestPayer', 'ExpectedBucketOwner']`if you think that could be useful.

Reply:
@zendesk-klross We can make this PR specific to encryption to keep the testing burden low for now if there aren't any other motivating use cases.

Reply:
Would you be able to create a version that just handles that so that we can kick off the review process?

Reply:
@savingoyal I've just opened a new PR [https://github.com/Netflix/metaflow/pull/1436](url) that limits the scope of this to server-side-encryption, let me know what you think!